src/py/flwr/simulation/ray_transport/ray_client_proxy.py

```diff
@@ -115,6 +116,117 @@ def reconnect(
         return common.DisconnectRes(reason="")  # Nothing to do here (yet)
 
 
+class RayActorClientProxy(ClientProxy):
+    """Flower client proxy which delegates work using Ray."""
+
+    def __init__(
+        self, client_fn: ClientFn, cid: str, actor_pool: VirtualClientEngineActorPool
+    ):
+        super().__init__(cid)
+        self.client_fn = client_fn
+        self.actor_pool = actor_pool
+
+    def _submit_job(self, job: Callable, timeout: Optional[float]):
+        try:
+            self.actor_pool.submit_client_job(
+                lambda a, v: a.run.remote(v, self.cid), job, self.cid
+            )
+            res = self.actor_pool.get_client_result(self.cid, timeout)
+
+        except Exception as ex:
+            if self.actor_pool.num_actors == 0:
+                # At this point we want to stop the simulation.
+                # since no more client workloads will be executed
+                log(ERROR, "ActorPool is empty!!! Disconnecting VirtualClient")
+                # TODO: the below does nothing?
+                self.reconnect()
```

Comment:
Is it possible that more actors are available at a later point?

Reply:
could be if the user adds a new node to Ray. this is a TODO (mentioned at the end of `VirtualClientEngineActorPool` constructor.

Reply:
yes. this has been added using a `threading.Timer` outside the actorpool. I think it's better to expand the pool from outside the object.

Reply:
yes. this is now supported.